src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -0,0 +1,126 @@
+#include <print.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0;} } while(0)
+
+#include <cos_alloc.h>
+#include <cobj_format.h>
+#include <cos_types.h>
+
+/* Assembly function for sinv from new component */
+extern void *__inv_test_entry(int a, int b, int c);
+
+struct cobj_header *hs[MAX_NUM_SPDS+1];
+
+/* The booter uses this to keep track of each comp */
+struct comp_cap_info {
+	struct cos_compinfo *compinfo;
+	vaddr_t addr_start;
+	vaddr_t vaddr_mapped_in_booter;
+	vaddr_t upcall_entry;
+} new_comp_cap_info[MAX_NUM_SPDS+1];
+
+struct cos_compinfo boot_info;
+struct cos_compinfo new_compinfo[MAX_NUM_SPDS+1];
+
+thdcap_t schedule[MAX_NUM_SPDS+1];
+unsigned int sched_cur;
+
+/* Macro for sinv back to booter from new component */
+enum {
+	BOOT_SINV_CAP = round_up_to_pow2(BOOT_CAPTBL_FREE + CAP32B_IDSZ, CAPMAX_ENTRY_SZ)
+};
+
+static vaddr_t
+boot_comp_mmap(int spdid, vaddr_t dest_daddr)
+{
+	vaddr_t addr = cos_page_bump_alloc(&boot_info);
+	assert(addr);
+			
+	if (cos_mem_alias_at(new_comp_cap_info[spdid].compinfo, dest_daddr, &boot_info, addr)) BUG();
+
+	return addr;
+}
+
+static void
+boot_comp_pgtbl_expand(int n_pte, pgtblcap_t pt, vaddr_t vaddr)
+{
+	int j;	
+	for (j = 0 ; j < n_pte ; j++) {
+		if (!cos_pgtbl_intern_alloc(&boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
+	}
+}
+
+
+/* Initialize just the captblcap and pgtblcap, due to hack for upcall_fn addr */
+static void
+boot_compinfo_init(int spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
+{
+	*ct = cos_captbl_alloc(&boot_info);
+	assert(*ct);
+	*pt = cos_pgtbl_alloc(&boot_info);
+	assert(*pt);
+	new_comp_cap_info[spdid].compinfo = &new_compinfo[spdid];
+	cos_compinfo_init(new_comp_cap_info[spdid].compinfo, *pt, *ct, 0, 
+				  (vaddr_t)vaddr, 4, &boot_info);
+}
+
+static void
+boot_newcomp_create(int spdid, captblcap_t ct, pgtblcap_t pt)
```

Comment:
kinda crappy that we have to pass the `*cap_t` stuff here.  Would be better if we could just pass the `comp_cap_info` struct as that abstracts these details.  The cap stuff is specific to this booter.  A cbuf booter might actually invoke another component to create the new component, which means passing caps here is not appropriate.  

We'll reach a pointer where we just need to get this done without everything being properly abstracted, but this seems like an easy modification.